invokeai/app/invocations/prompt.py

```diff
@@ -55,3 +56,35 @@ def invoke(self, context: InvocationContext) -> PromptCollectionOutput:
             prompts = generator.generate(self.prompt, num_images=self.max_prompts)
 
         return PromptCollectionOutput(prompt_collection=prompts, count=len(prompts))
+    
+
+class PromptsFromFileInvocation(BaseInvocation):
+    '''Loads prompts from a text file'''
+    # fmt: off
+    type: Literal['prompt_from_file'] = 'prompt_from_file'
+
+    # Inputs
+    filename: str = Field(default=None, description="Filename of prompt text file")
+    pre_prompt: str = Field(default=None, description="Add to start of prompt")
+    post_prompt: str = Field(default=None, description="Add to end of prompt")
+    start_line: int = Field(default=1, ge=1, description="Line in the file start start from")
+    max_prompts: int = Field(default=0, ge=0, description="Max lines to read from file (0=all)")
+    #fmt: on
+
+    def promptsFromFile(self, filename: str, pre_prompt: str, post_prompt: str, start_line: int, max_prompts:int):
```

Comment:
Given the needed changes to types above, `pre_prompt` and `post_prompt` will need to be typed differently.

When this function is called, they will either be `str` or `None`. This can be represented with `Union[str, None]`. There's a simpler `Union` syntax (`str | None`) but it's not supported on python 3.9, which we still support, so you need to use the bracket syntax.